src/plots/mapbox/layout_attributes.js

```diff
@@ -48,10 +48,16 @@ var attrs = module.exports = overrideAll({
         dflt: constants.styleValueDflt,
         role: 'style',
         description: [
-            'Sets the Mapbox map style.',
-            'Either input one of the default Mapbox style names or the URL to a custom style',
-            'or a valid Mapbox style JSON.',
-            'From OpenStreetMap raster tiles, use *open-street-map*.'
+            'Sets the Mapbox base map style.',
+            'Base map styles are rendered below all traces and layout layers.',
+            'Either input one of the default Mapbox style names:', constants.styleValuesMapbox, '.',
+            'Note that to use these, a Mapbox access token must be set either in the `accesstoken` attribute',
+            'or in the `mapboxAccessToken` config option.',
+            'From OpenStreetMap raster tiles, use:', constants.styleValueOSM, '.',
+            'No access token is required to render the', constants.styleValueOSM, 'style.',
+            'One can also set `style` as a URL to a Mapbox custom style, e.g. created in Mapbox Studio.',
+            'Finally, one can set `style` as a Mapbox style JSON, see',
```

Comment:
```suggestion
            'Finally, one can set `style` as a Mapbox style JSON, see ',
```

Reply:
nvm - i'm assuming a space is added after every line

Reply:
I would reverse the order in which we explain this: not [name, url, json] but rather [json, osm, url, name].

My reasoning is that really the intent of this attribute is to specify or identify the JSON object that will be used to render the map. So we should say that first. Then we should say that we provide a shortcut to a particular OSM-based one which you can pass in as a string. Then we should say that you can create and host these styles in the Mapbox studio app, in which case you can provide the URL here and you'll need a token. Then finally we should say that Mapbox provides some default hosted styles and for convenience we have shortcuts for these enumerated things, which are expanded to URLs by prepending `"mapbox://styles/mapbox/"` or whatever. 

I would draw a distinction between the Mapbox-service-provided shortcut names and our new plotly.js-provided shortcut name just to be clear. For example until reading the actual source of https://github.com/plotly/plotly.js/blob/c1ef6911da054f3b16a7abe8fb2d56019988ba14/src/plots/mapbox/constants.js it wasn't clear to me that the "basic", "light", "dark" etc were not specified in our source tree.

Reply:
> My reasoning is that really the intent of this attribute is to specify or identify the JSON object that will be used to render the map. So we should say that first

I disagree here. As long as our default `mapbox.style` is a Mapbox-service style, I think it makes more sense to advertise the other available Mapbox-service style values first. This is of course subject to change when/if we make `'open-street-map'` the default in a major version bump. I really can't imagine that many users playing around with the `mapbox.style` JSON structure. That said, you have veto rights @nicolaskruchten - just writing down my opinion here.


-------------

> I would draw a distinction between the Mapbox-service-provided shortcut names and our new plotly.js-provided shortcut name just to be clear

I don't fully understand what we mean here. Would turning:

```
            'For OpenStreetMap raster tiles, use: ', constants.styleValueOSM, '.',
            'No access token is required to render the', constants.styleValueOSM, 'style.',
```

into 

```
            'For OpenStreetMap raster tiles, use the Plotly shortcut value: ', constants.styleValueOSM, '.',
            'No access token is required to render the', constants.styleValueOSM, 'style.',
```

be ok?

Reply:
> I disagree here. As long as our default mapbox.style is a Mapbox-service style,

Our intent with 1.49 is to permit non-Mapbox, non-OSM layers to be specified as the lowest "base" layers... The only way to do this is to specify this JSON inline. This needs to be more clearly documented front and center IMO.

I also still think that `open-street-map` should be the default if no mapbox access token is present.

Reply:
> I also still think that open-street-map should be the default if no mapbox access token is present.

I'd call that _too magical_.

Reply:
Why? Right now if you don't give a token, you get nothing. Something is better than nothing.

Reply:
I'm not sure which side I fall on here in the end, but let's say a user wants mapbox, they make a plot but forget to provide their token, and it shows up as OSM... how do they debug this? How do they figure out that it's the lack of token that gave them OSM? Feels like "you get nothing" is a clearer signal that you just omitted something, whereas "you get OSM" sends you down the path of trying to find where exactly you asked for OSM - answer: nowhere, but if you're lucky you'll find that the docs for which provider to use tell you about the token via their description of the default. Can we improve on this experience?

Reply:
OK, that's an actual objection I can respond to :)

In this case I think that I would say that if neither the token nor the style are provided, the console error message that currently says "token missing" should say "please specify a style". Instead of the current behaviour which automatically encourages mapbox without leaving the door open to non-mapbox styles. The current message is appropriate when the style is a mapbox style and no token is provided because then the user is explicitly requesting a mapbox style and should provide a token.

Reply:
(follow-up in streambed)